index.js

```diff
@@ -137,11 +138,14 @@ function createFocusTrap(element, userOptions) {
     // There can be only one listening focus trap at a time
     activeFocusTraps.activateTrap(trap);
 
+    var initialFocusNode = getInitialFocusNode();
     // Delay ensures that the focused element doesn't capture the event
     // that caused the focus trap activation.
-    activeFocusDelay = delay(function () {
-      tryFocus(getInitialFocusNode());
-    });
+    activeFocusDelay = config.delayInitialFocus
+      ? delay(function () {
+          tryFocus(initialFocusNode);
+        })
+      : tryFocus(initialFocusNode);
```

Comment:
While I definitely appreciate the optimization of calling `getInitialFocusNode()` once for either case of `delayInitialFocus`, and probably something I would have done myself, seeing it here is making me realize that the previous behavior was actually allowing the initial focus node to not exist in the current event loop pass that activates the trap because the call to `getInitialFocusNode()` was also delayed to the pass. That could potentially allow for the initial focus node to be created in the same pass that activates the trap, after calling `activate()` on the trap.

__TL/DR__: So just to be extra cautious, I'd prefer _not_ to optimize here, and call `getInitialFocusNode()` in both places rather than using the variable approach.